api/migrations/0003_fake_data.py

```diff
@@ -278,11 +278,16 @@ def _make_amount_of_recipe_and_skip_database_error(
         """
         for _ in range(amount):
             while True:
+                # Violation of CHECK OR UNIQUE constraint will trigger IntegrityError
                 try:
-                    recipe.make()
-                    break
-                except DatabaseError:
-                    pass
+                    # context manager has to be used to avoid TransactionManagementError.
+                    # See section "Controlling transactions explicitly" in
+                    # https://docs.djangoproject.com/en/dev/topics/db/transactions/
+                    with transaction.atomic():
+                        recipe.make()
+                        break
+                except IntegrityError as e:
```

Comment:
IntegrityError is a subclass of DatabaseError, so I think in previous version it also should work 🤔

Reply:
OK, but the main thing is that I added `transaction.atomic()`

Reply:
Ah, ok. But probably let's keep DatabaseError to avoid any other db exception?

Reply:
Done